gems/smithy-client/spec/smithy-client/endpoint_rules_spec.rb

```diff
@@ -0,0 +1,202 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    describe EndpointRules do
+      describe '.valid_host_label?' do
```

Comment:
Do we not cover these tests from standard endpoint test rules? I thought most of these are defined in a json somewhere.

Reply:
These do end up getting tested through endpoint rules - but to have coverage in this library, we need the specs (and we had them in hearth, I just copied them over).

Reply:
That's strange - are we sure they were even tested through endpoint rules?

Reply:
Some of them are better tested than others through test rule files.  Its one of my long term goals (when I get extra time...) to really improve those shared endpoint test cases.